app/controllers/users_controller.rb

```diff
@@ -0,0 +1,19 @@
+class UsersController < ApplicationController
+  get '/users/new' do
+    email = session[:sign_up_email] || ""
+    session[:sign_up_email] = nil
+    erb :"users/new.html", locals: {email: email}
+  end
+
+  post '/users' do
+    user = User.new(email: params[:email], password: params[:password], password_confirmation: params[:password_confirmation])
+    session[:sign_up_email] = nil
```

Comment:
Rather than storing the email they put in the form in the `session`, you can pass the `user` object into the `erb` template via. `locals`